backend/song/tests/test_views.py

```diff
@@ -0,0 +1,451 @@
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from backend.users.models import User
+
+from backend.song.models import Artist
+
+
+class BaseTestSetUp(APITestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            username="test",
+            email="test@nzembo.msc",
+            name="test user",
+            password="strongP@sswor2"
+        )
+
+        self.authenticate()
+
+        self.artistInfo = {
+            'first_name': 'Jules',
+            'last_name': 'Shungu',
+            'is_still_alive': False,
+            'image': 'https://lastfm.freetls.fastly.net/i/u/arO/7de29a9d31b55e3ee367cecf152c0052'
+        }
+        self.artist = Artist.objects.create(**self.artistInfo)
+
+        self.albumInfo = {
+            'album_name': 'tango to zalaki',
+            'artist': self.artist.id
+        }
+
+        self.language = {
+            'language': 'french'
+        }
+
+        self.song = {
+            'title': 'maboko pamba',
+            'category': 1,
+            'album': 1,
+            'artist': self.artist.id
+        }
+
+        self.lyric = {
+            'lyric': "What is Lorem Ipsum Lorem Ipsum is simply dummy text of the printing",
+            'song': 1,
+            'language': 1
+        }
+
+        self.lyric_request = {
+            'song': 1,
+            'language': 1,
+            'message': "I need a lyric for this song"
+        }
+
+        self.translation = {
+            'lyric': 1,
+            'language': 1,
+            'translation': "My translation"
+        }
+
+    def authenticate(self):
+        self.client.force_authenticate(user=self.user)
+
+    def logout(self):
+        self.client.logout()
+
+    def send_a_single_category_post_request(self):
+        return self.client.post('/api/v1/categories/', data={'category_name': 'rumba'}, format='json')
+
+    def send_a_single_album_post_request(self):
+        return self.client.post('/api/v1/albums/', data=self.albumInfo, format='json')
+
+    def send_a_single_language_post_request(self):
+        return self.client.post('/api/v1/languages/', data=self.language, format='json')
+
+    def send_a_single_song_post_request(self):
+        return self.client.post('/api/v1/songs/', data=self.song, format='json')
+
+    def send_a_single_lyrirc_post_request(self):
+        return self.client.post('/api/v1/lyrics/', data=self.lyric, format='json')
+
+    def send_a_single_lyric_request_post_request(self):
+        return self.client.post('/api/v1/lyric-requests/', data=self.lyric_request, format='json')
+
+    def send_a_single_translation_post_request(self):
+        return self.client.post('/api/v1/translations/', data=self.translation, format='json')
```

Comment:
if you add names in the URLs declaration you will not need to hardcode them here...

you can just call them with their names using `reverse` function...
more info [here](https://www.webforefront.com/django/namedjangourls.html)